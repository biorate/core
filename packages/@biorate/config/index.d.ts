import { PropertyPath } from 'lodash';
import { BaseError } from '@biorate/errors';
import { IConfig } from './interfaces';

declare module '@biorate/config' {
  export type IConfig = IConfig;

  export class Config {
    protected data: Record<string | symbol, unknown>;

    protected template(value: string): string;

    protected templatize(object: unknown): any;

    public get<T = unknown>(path: PropertyPath, def?: T): T;

    public has(path: PropertyPath): boolean;

    public set(path: PropertyPath, value: unknown): void;

    public merge(data: unknown): void;
  }

  export class UndefinedConfigPathError extends BaseError {
    public constructor(path: PropertyPath);
  }
}
