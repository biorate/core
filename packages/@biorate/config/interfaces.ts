import { PropertyPath } from 'lodash';

export interface IConfig {
  get<T = unknown>(path: PropertyPath, def?: T): T;
  has(path: PropertyPath): boolean;
  set(path: PropertyPath, value: unknown): void;
  merge(data: unknown): void;
}
