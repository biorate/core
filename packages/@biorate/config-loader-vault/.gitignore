# compiled output
dist
node_modules
cache
keys

# Logs
logs
*.log
npm-debug.log*
yarn-debug.log*
yarn-error.log*
lerna-debug.log*

# Diagnostic reports (https://nodejs.org/api/report.html)
report.[0-9]*.[0-9]*.[0-9]*.[0-9]*.json

# OS
.DS_Store

# Coverage directory used by tools like istanbul
coverage
*.lcov

# nyc test coverage
.nyc_output

# Optional npm cache directory
.npm

# Optional eslint cache
.eslintcache

# IDEs and editors
.idea
.project
.classpath
.c9/
*.launch
.settings/
*.sublime-workspace

# IDE - VSCode
.vscode/*
!.vscode/settings.json
!.vscode/tasks.json
!.vscode/launch.json
!.vscode/extensions.json

# Package lock
package-lock.json
