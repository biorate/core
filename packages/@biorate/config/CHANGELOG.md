# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.24.0](https://github.com/biorate/core/compare/v1.23.2...v1.24.0) (2023-02-01)

**Note:** Version bump only for package @biorate/config





## [1.11.2](https://github.com/biorate/core/compare/v1.11.1...v1.11.2) (2022-09-21)


### Bug Fixes

* **axios, kafka:** logic fixes ([5b83869](https://github.com/biorate/core/commit/5b8386953ce1d63783f380ca8fc36fb324a963b1))





## [1.11.1](https://github.com/biorate/core/compare/v1.11.0...v1.11.1) (2022-09-21)


### Bug Fixes

* **axios, kafka:** logic fixes ([d48fff0](https://github.com/biorate/core/commit/d48fff0ebc75566b59aff08e17372ac97b595b30))





# [1.11.0](https://github.com/biorate/core/compare/v1.10.0...v1.11.0) (2022-09-21)

**Note:** Version bump only for package @biorate/config





# [1.7.0](https://github.com/biorate/core/compare/v1.6.0...v1.7.0) (2022-09-08)


### Features

* **config:** link template ([b1ea054](https://github.com/biorate/core/commit/b1ea054ebe6953e0fb7073cc1c6b273df06a7cf3))





# [1.6.0](https://github.com/biorate/core/compare/v1.5.1...v1.6.0) (2022-09-08)


### Features

* **config-loader:** concept added ([0e58d55](https://github.com/biorate/core/commit/0e58d5509323aa0a0a7689642708788e2188db7b))





# [1.4.0](https://github.com/biorate/core/compare/v1.3.2...v1.4.0) (2022-08-15)

**Note:** Version bump only for package @biorate/config





## [1.2.2](https://github.com/biorate/core/compare/v1.2.1...v1.2.2) (2022-08-11)

**Note:** Version bump only for package @biorate/config





# [1.0.0](https://github.com/biorate/core/compare/v0.30.19...v1.0.0) (2022-07-10)

**Note:** Version bump only for package @biorate/config





## [0.30.19](https://github.com/biorate/core/compare/v0.30.18...v0.30.19) (2022-07-10)

**Note:** Version bump only for package @biorate/config





## [0.30.11](https://github.com/biorate/core/compare/v0.30.10...v0.30.11) (2022-06-24)


### Bug Fixes

* **config:** npmignore fix ([1ef9a50](https://github.com/biorate/core/commit/1ef9a509fc5764dd6111e4fec0fbe9ae7f2e2a6f))





## [0.30.10](https://github.com/biorate/core/compare/v0.30.9...v0.30.10) (2022-06-24)


### Bug Fixes

* **config:** strict types ([5c4a1fd](https://github.com/biorate/core/commit/5c4a1fd6f8008538fb13f6a303909730f7bc2d31))





## [0.30.9](https://github.com/biorate/core/compare/v0.30.8...v0.30.9) (2022-06-23)


### Bug Fixes

* **config:** strict types ([d1e286e](https://github.com/biorate/core/commit/d1e286e9016938cb663ee8bcceb6c06b0dbfdb18))





## [0.30.7](https://github.com/biorate/core/compare/v0.30.6...v0.30.7) (2022-06-23)

**Note:** Version bump only for package @biorate/config





## [0.30.6](https://github.com/biorate/core/compare/v0.30.5...v0.30.6) (2022-06-23)

**Note:** Version bump only for package @biorate/config





## [0.30.5](https://github.com/biorate/core/compare/v0.30.4...v0.30.5) (2022-06-23)

**Note:** Version bump only for package @biorate/config





## [0.30.4](https://github.com/biorate/core/compare/v0.30.3...v0.30.4) (2022-06-23)

**Note:** Version bump only for package @biorate/config





## [0.30.2](https://github.com/biorate/core/compare/v0.30.1...v0.30.2) (2022-06-22)

**Note:** Version bump only for package @biorate/config





# [0.30.0](https://github.com/biorate/core/compare/v0.29.1...v0.30.0) (2022-06-20)


### Bug Fixes

* *.ts into npmignore ([6a6be03](https://github.com/biorate/core/commit/6a6be030a1e8923df4b03d518f08618e759d06fa))





## [0.29.1](https://github.com/biorate/core/compare/v0.29.0...v0.29.1) (2022-06-20)

**Note:** Version bump only for package @biorate/config





# [0.29.0](https://github.com/biorate/core/compare/v0.28.3...v0.29.0) (2022-06-20)


### Bug Fixes

* *.ts into npmignore ([f6e68b3](https://github.com/biorate/core/commit/f6e68b3a2291e7bfa013674db857db7879a26033))





# [0.28.0](https://github.com/biorate/core/compare/v0.27.3...v0.28.0) (2022-05-06)

**Note:** Version bump only for package @biorate/config





# [0.27.0](https://github.com/biorate/core/compare/v0.26.0...v0.27.0) (2022-05-06)

**Note:** Version bump only for package @biorate/config





# [0.26.0](https://github.com/biorate/core/compare/v0.25.0...v0.26.0) (2022-04-25)

**Note:** Version bump only for package @biorate/config





# [0.25.0](https://github.com/biorate/core/compare/v0.24.0...v0.25.0) (2022-04-25)


### Features

* **schema-registry:** api fix ([8631389](https://github.com/biorate/core/commit/8631389036e4c0d1f0a9ec0ccab5a0a2800c4dab))


### BREAKING CHANGES

* **schema-registry:** api fix according axios fatch interface change





# [0.24.0](https://github.com/biorate/core/compare/v0.23.4...v0.24.0) (2022-04-25)

**Note:** Version bump only for package @biorate/config





# [0.23.0](https://github.com/biorate/core/compare/v0.22.0...v0.23.0) (2022-04-11)

**Note:** Version bump only for package @biorate/config





# [0.22.0](https://github.com/biorate/core/compare/v0.21.0...v0.22.0) (2022-04-08)

**Note:** Version bump only for package @biorate/config





# [0.21.0](https://github.com/biorate/core/compare/v0.20.0...v0.21.0) (2022-04-08)

**Note:** Version bump only for package @biorate/config





# [0.20.0](https://github.com/biorate/core/compare/v0.19.1...v0.20.0) (2022-04-08)

**Note:** Version bump only for package @biorate/config





# [0.19.0](https://github.com/biorate/core/compare/v0.18.2...v0.19.0) (2022-04-07)

**Note:** Version bump only for package @biorate/config





## [0.18.2](https://github.com/biorate/core/compare/v0.18.1...v0.18.2) (2022-03-18)

**Note:** Version bump only for package @biorate/config





# [0.18.0](https://github.com/biorate/core/compare/v0.17.1...v0.18.0) (2022-02-05)

**Note:** Version bump only for package @biorate/config





## [0.17.1](https://github.com/biorate/core/compare/v0.17.0...v0.17.1) (2022-02-04)


### Bug Fixes

* **config:** constructor tests fix ([7fe53b4](https://github.com/biorate/core/commit/7fe53b48d127dbcbf8f0cc7cc448a85819ef56ae))





# [0.17.0](https://github.com/biorate/core/compare/v0.16.3...v0.17.0) (2022-02-04)


### Features

* **config:** remove constructor logic, add injectable ([69391d9](https://github.com/biorate/core/commit/69391d9aa250dd87b734df6e3d0a6b7937a2e69b))





## [0.16.1](https://github.com/biorate/core/compare/v0.16.0...v0.16.1) (2022-02-01)


### Bug Fixes

* publishConfig added in package.json ([716a064](https://github.com/biorate/core/commit/716a06429e1a4325f50cf94b804f1d70d8a58cf5))





# [0.16.0](https://github.com/biorate/core/compare/v0.15.2...v0.16.0) (2022-02-01)


### Bug Fixes

* **config:** IConfig interface added ([b6e6bf1](https://github.com/biorate/core/commit/b6e6bf1fe267f0d87ad69581698ea72c90f5969d))





# [0.15.0](https://github.com/biorate/core/compare/v0.14.1...v0.15.0) (2022-01-31)

**Note:** Version bump only for package @biorate/config





# [0.14.0](https://github.com/biorate/core/compare/v0.13.0...v0.14.0) (2021-12-31)

**Note:** Version bump only for package @biorate/config





# [0.13.0](https://github.com/biorate/core/compare/v0.12.4...v0.13.0) (2021-12-31)

**Note:** Version bump only for package @biorate/config





## [0.12.4](https://github.com/biorate/core/compare/v0.12.3...v0.12.4) (2021-12-25)

**Note:** Version bump only for package @biorate/config





## [0.12.3](https://github.com/biorate/core/compare/v0.12.2...v0.12.3) (2021-12-25)

**Note:** Version bump only for package @biorate/config





## [0.12.2](https://github.com/biorate/core/compare/v0.12.1...v0.12.2) (2021-12-25)


### Bug Fixes

* **all:** homepage and repository in package.json added ([532c5a0](https://github.com/biorate/core/commit/532c5a08ed8cd9c1590134699c074c2021bcbbff))





## [0.12.1](https://github.com/biorate/core/compare/v0.12.0...v0.12.1) (2021-12-25)

**Note:** Version bump only for package @biorate/config





# [0.12.0](https://github.com/biorate/core/compare/v0.11.2...v0.12.0) (2021-12-25)

**Note:** Version bump only for package @biorate/config





## [0.11.1](https://github.com/biorate/core/compare/v0.11.0...v0.11.1) (2021-10-16)

**Note:** Version bump only for package @biorate/config





# [0.11.0](https://github.com/biorate/core/compare/v0.10.4...v0.11.0) (2021-10-16)

**Note:** Version bump only for package @biorate/config





## [0.10.2](https://github.com/biorate/core/compare/v0.10.1...v0.10.2) (2021-09-27)


### Bug Fixes

* **config:** d.ts annotation ([29e65c9](https://github.com/biorate/core/commit/29e65c9d12c0bb13f54a1ca34128d53c5b5b006d))





## [0.10.1](https://github.com/biorate/core/compare/v0.10.0...v0.10.1) (2021-09-27)

**Note:** Version bump only for package @biorate/config





# [0.10.0](https://github.com/biorate/core/compare/v0.9.3...v0.10.0) (2021-09-24)


### Features

* **config:** release ([f7ba2b9](https://github.com/biorate/core/commit/f7ba2b91a8a5c766c4440dfc0243d42a7cc833b9))
* **config:** release ([0fce238](https://github.com/biorate/core/commit/0fce238c553057a6eec7fa3ddb2e2448d79eb3d2))
