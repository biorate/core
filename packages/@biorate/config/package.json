{
  "name": "@biorate/config",
  "version": "1.91.2",
  "description": "Application configurator",
  "main": "dist",
  "scripts": {
    "build": "npx tsc -p ./tsconfig.build.json --outDir ./dist",
    "test": "npx nyc --reporter=lcov --reporter=text-summary -- npx mocha -r ts-node/register tests/**/*.spec.ts",
    "prepublishOnly": "npm run build"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/biorate/core.git",
    "directory": "packages/@biorate/config"
  },
  "publishConfig": {
    "access": "public"
  },
  "homepage": "https://biorate.github.io/core",
  "keywords": [],
  "author": "llevkin",
  "license": "MIT",
  "dependencies": {
    "@biorate/errors": "1.28.0",
    "@biorate/inversion": "1.65.4",
    "lodash": "^4.17.21",
    "traverse": "^0.6.6"
  },
  "gitHead": "fdd6dbd61368f7a9f9deb3b72243b19ecb3767a6",
  "devDependencies": {
    "@types/traverse": "^0.6.32"
  }
}
