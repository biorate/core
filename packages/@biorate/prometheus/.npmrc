registry=https://art.lmru.tech/api/npm/npm-payment/
