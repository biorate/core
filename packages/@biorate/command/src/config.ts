import { Config as CommonConfig } from '@biorate/config';

export class Config extends CommonConfig {
  public all(): Record<string, any> {
    return this.templatize(this.data);
  }

  public clear() {
    this.data = {};
  }
}
