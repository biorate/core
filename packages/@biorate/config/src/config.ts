import { injectable } from '@biorate/inversion';
import { merge, get, has, set, PropertyPath } from 'lodash';
import * as traverse from 'traverse';
import { Template } from './template';
import { UndefinedConfigPathError } from './errors';
import { IConfig, IResult } from '../interfaces';

@injectable()
export class Config implements IConfig {
  /**
   * @description Template enabling settings
   * */
  public static readonly Template = {
    string: true,
    link: true,
    regexp: true,
    function: true,
    empty: true,
  };

  /**
   * @description Data storage
   * */
  protected data = {};

  /**
   * @description Set template value
   * */
  protected template<T = unknown>(value: string, def?: T) {
    const result: IResult = { value };
    Template.string.call(this, value, result, def);
    Template.link.call(this, value, result, def);
    Template.regexp.call(this, value, result, def);
    Template.function.call(this, value, result, def);
    Template.empty.call(this, value, result, def);
    return result.value;
  }

  /**
   * @description Walk object recursive to templatize internal values
   * */
  protected templatize<T = unknown>(object: unknown, def?: T) {
    const template = this.template.bind(this);
    return traverse(object).forEach(function (value) {
      if (typeof value === 'string') this.update(template(value));
    });
  }

  /**
   * @description Get config property by path
   * @param path - data path
   * @param def - default value
   * @example
   * ```
   * import { Config } from '@biorate/config';
   *
   * const config = new Config();
   *
   * config.set('a', 1);
   *
   * console.log(config.get('a')); // 1
   * console.log(config.get('b', 2)); // 2
   * console.log(config.get('b')); // UndefinedConfigPathError: Undefined config path [b]
   *                               //  at Config.get (src/index.ts:2:1608)
   *                               //  at Context.<anonymous> (tests/index.spec.ts:19:24)
   *                               //  at processImmediate (node:internal/timers:464:21)
   *
   * ```
   * */
  public get<T = unknown>(path: PropertyPath, def?: T): T {
    if (!this.has(path)) {
      if (def === undefined) throw new UndefinedConfigPathError(path);
      return def;
    }
    const result: any = get(this.data, path);
    switch (typeof result) {
      case 'string':
        return <T>(<unknown>this.template<T>(result, def));
      case 'object':
        return result ? this.templatize<T>(result, def) : result;
      default:
        return result;
    }
  }

  /**
   * @description Has config property by path
   * @param path - data path
   * @example
   * ```
   * import { Config } from '@biorate/config';
   *
   * const config = new Config();
   *
   * config.set('a', 1);
   *
   * console.log(config.has('a')); // true
   * ```
   * */
  public has(path: PropertyPath) {
    return has(this.data, path);
  }

  /**
   * @description Set config property by path
   * @param path - data path
   * @param value - data value
   * @example
   * ```
   * import { Config } from '@biorate/config';
   *
   * const config = new Config();
   *
   * config.set('a', 1);
   *
   * console.log(config.get('a')); // 1
   * ```
   * */
  public set(path: PropertyPath, value: unknown) {
    set(this.data, path, value);
  }

  /**
   * @description Merge config data
   * @param data - data object
   * @example
   * ```
   * import { Config } from '@biorate/config';
   *
   * const config = new Config();
   *
   * config.merge({
   *   a: { b: { c: 1 } },
   * });
   *
   * config.merge({
   *   a: { b: { d: 2 } },
   * });
   *
   * console.log(config.has('a')); // true
   * console.log(config.has('a.b')); // true
   * console.log(config.get('a.b.c')); // 1
   * console.log(config.get('a.b.d')); // 2
   * ```
   * */
  public merge(data: unknown) {
    merge(this.data, data);
  }
}
