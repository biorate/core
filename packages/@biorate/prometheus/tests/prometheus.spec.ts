import { expect } from 'chai';
import { container } from '@biorate/inversion';
// import { Prometheus } from '../';
import { Root } from './index.config';

describe('Prometheus', function () {
  let root: Root;
  this.timeout(3e4);

  before(async () => {
    root = container.get<Root>(Root);
    await root.$run();
  });

  it('counter', async () => {
    root.counter.inc();
    expect(await root.prometheus.registry.metrics()).toMatchSnapshot();
  });

  it('gauge', async () => {
    root.gauge.inc();
    expect(await root.prometheus.registry.metrics()).toMatchSnapshot();
  });

  it('summary', async () => {
    root.summary.observe(1);
    expect(await root.prometheus.registry.metrics()).toMatchSnapshot();
  });

  it('histogram', async () => {
    root.histogram.observe(1);
    expect(await root.prometheus.registry.metrics()).toMatchSnapshot();
  });
});
