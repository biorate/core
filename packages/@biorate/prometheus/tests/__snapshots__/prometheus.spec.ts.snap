// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prometheus counter 1`] = `
"# HELP counter counter
# TYPE counter counter
counter 1

# HELP gauge gauge
# TYPE gauge gauge
gauge 0

# HELP summary summary
# TYPE summary summary
summary{quantile=\\"0.01\\"} 0
summary{quantile=\\"0.05\\"} 0
summary{quantile=\\"0.5\\"} 0
summary{quantile=\\"0.9\\"} 0
summary{quantile=\\"0.95\\"} 0
summary{quantile=\\"0.99\\"} 0
summary{quantile=\\"0.999\\"} 0
summary_sum 0
summary_count 0

# HELP histogram histogram
# TYPE histogram histogram
histogram_bucket{le=\\"0.005\\"} 0
histogram_bucket{le=\\"0.01\\"} 0
histogram_bucket{le=\\"0.025\\"} 0
histogram_bucket{le=\\"0.05\\"} 0
histogram_bucket{le=\\"0.1\\"} 0
histogram_bucket{le=\\"0.25\\"} 0
histogram_bucket{le=\\"0.5\\"} 0
histogram_bucket{le=\\"1\\"} 0
histogram_bucket{le=\\"2.5\\"} 0
histogram_bucket{le=\\"5\\"} 0
histogram_bucket{le=\\"10\\"} 0
histogram_bucket{le=\\"+Inf\\"} 0
histogram_sum 0
histogram_count 0
"
`;

exports[`Prometheus gauge 1`] = `
"# HELP counter counter
# TYPE counter counter
counter 1

# HELP gauge gauge
# TYPE gauge gauge
gauge 1

# HELP summary summary
# TYPE summary summary
summary{quantile=\\"0.01\\"} 0
summary{quantile=\\"0.05\\"} 0
summary{quantile=\\"0.5\\"} 0
summary{quantile=\\"0.9\\"} 0
summary{quantile=\\"0.95\\"} 0
summary{quantile=\\"0.99\\"} 0
summary{quantile=\\"0.999\\"} 0
summary_sum 0
summary_count 0

# HELP histogram histogram
# TYPE histogram histogram
histogram_bucket{le=\\"0.005\\"} 0
histogram_bucket{le=\\"0.01\\"} 0
histogram_bucket{le=\\"0.025\\"} 0
histogram_bucket{le=\\"0.05\\"} 0
histogram_bucket{le=\\"0.1\\"} 0
histogram_bucket{le=\\"0.25\\"} 0
histogram_bucket{le=\\"0.5\\"} 0
histogram_bucket{le=\\"1\\"} 0
histogram_bucket{le=\\"2.5\\"} 0
histogram_bucket{le=\\"5\\"} 0
histogram_bucket{le=\\"10\\"} 0
histogram_bucket{le=\\"+Inf\\"} 0
histogram_sum 0
histogram_count 0
"
`;

exports[`Prometheus histogram 1`] = `
"# HELP counter counter
# TYPE counter counter
counter 1

# HELP gauge gauge
# TYPE gauge gauge
gauge 1

# HELP summary summary
# TYPE summary summary
summary{quantile=\\"0.01\\"} 1
summary{quantile=\\"0.05\\"} 1
summary{quantile=\\"0.5\\"} 1
summary{quantile=\\"0.9\\"} 1
summary{quantile=\\"0.95\\"} 1
summary{quantile=\\"0.99\\"} 1
summary{quantile=\\"0.999\\"} 1
summary_sum 1
summary_count 1

# HELP histogram histogram
# TYPE histogram histogram
histogram_bucket{le=\\"0.005\\"} 0
histogram_bucket{le=\\"0.01\\"} 0
histogram_bucket{le=\\"0.025\\"} 0
histogram_bucket{le=\\"0.05\\"} 0
histogram_bucket{le=\\"0.1\\"} 0
histogram_bucket{le=\\"0.25\\"} 0
histogram_bucket{le=\\"0.5\\"} 0
histogram_bucket{le=\\"1\\"} 1
histogram_bucket{le=\\"2.5\\"} 1
histogram_bucket{le=\\"5\\"} 1
histogram_bucket{le=\\"10\\"} 1
histogram_bucket{le=\\"+Inf\\"} 1
histogram_sum 1
histogram_count 1
"
`;

exports[`Prometheus summary 1`] = `
"# HELP counter counter
# TYPE counter counter
counter 1

# HELP gauge gauge
# TYPE gauge gauge
gauge 1

# HELP summary summary
# TYPE summary summary
summary{quantile=\\"0.01\\"} 1
summary{quantile=\\"0.05\\"} 1
summary{quantile=\\"0.5\\"} 1
summary{quantile=\\"0.9\\"} 1
summary{quantile=\\"0.95\\"} 1
summary{quantile=\\"0.99\\"} 1
summary{quantile=\\"0.999\\"} 1
summary_sum 1
summary_count 1

# HELP histogram histogram
# TYPE histogram histogram
histogram_bucket{le=\\"0.005\\"} 0
histogram_bucket{le=\\"0.01\\"} 0
histogram_bucket{le=\\"0.025\\"} 0
histogram_bucket{le=\\"0.05\\"} 0
histogram_bucket{le=\\"0.1\\"} 0
histogram_bucket{le=\\"0.25\\"} 0
histogram_bucket{le=\\"0.5\\"} 0
histogram_bucket{le=\\"1\\"} 0
histogram_bucket{le=\\"2.5\\"} 0
histogram_bucket{le=\\"5\\"} 0
histogram_bucket{le=\\"10\\"} 0
histogram_bucket{le=\\"+Inf\\"} 0
histogram_sum 0
histogram_count 0
"
`;
